feat: file domain

nmrusso opened the pull request.

nmrusso committed:
feat: file domain

ANGdesarrollo commented on src/File/Application/Handlers/RemoveFileHandler.ts:
```diff
@@ -0,0 +1,31 @@
+import ValidatedHandler from '@shared/Validations/ValidatedHandler';
+
+@CommandHandler(RemoveFileCommand)
+class RemoveFileHandler extends ValidatedHandler<RemoveFileCommand, any> implements ICommandHandler<RemoveFileCommand>
```
¿Aca no devuelve el objeto que borramos? es decir un IFileDomain en vez de un any?

ANGdesarrollo commented on src/File/Domain/Entities/IFileDomain.ts:
```diff
@@ -0,0 +1,53 @@
+export default IFileDomain;
+
+/**
+[
```
¿Este comentario tiene que quedar?

ANGdesarrollo commented on src/File/Infrastructure/Schemas/FileSchema.ts:
```diff
@@ -0,0 +1,44 @@
+  return this._id;
+});
+
+// FileSchema.pre('save', function(next: any)
```
¿Este codigo comentado esta bien?

ANGdesarrollo committed:
feat: configure minio and command to create bucket

ANGdesarrollo committed:
feat: add query params for file upload and public/private bucket handling

               Introduced support for , , and  query parameters during file upload. Updated payload and handler to manage public/private bucket differentiation and validation schema transform for these parameters. Removed unused interface declarations.

ANGdesarrollo committed:
feat: simplify bucket parameter assignment

Remove unnecessary string interpolation when assigning the Minio bucket parameter. This change makes the code cleaner and easier to read, reducing potential errors related to string formatting.

ANGdesarrollo committed:
feat: enhance file handling and schema validation

Refactored UpdateMultipartFileSchemaValidation and added UpdateBucketRepRequest interface to improve schema validation. Adjusted handlers to incorporate new validation logic and modified filesystem strategies to support public/private file differentiation.

ANGdesarrollo committed:
refactor: import order in RemoveFileHandler

Rearranged the import statements in RemoveFileHandler.ts for better code organization, ensuring consistency with the project's import order conventions. This refactor does not alter any functionality.

ANGdesarrollo committed:
feat: rebase conflicts
